consensus/core/src/network/mod.rs

```diff
@@ -29,13 +30,19 @@ pub(crate) mod epoch_filter;
 pub(crate) mod metrics;
 pub(crate) mod tonic_network;
 
+/// A stream of serialized blocks returned over the network.
+pub(crate) type BlockStream = Pin<Box<dyn Stream<Item = Bytes> + Send>>;
+
 /// Network client for communicating with peers.
 ///
 /// NOTE: the timeout parameters help saving resources at client and potentially server.
 /// But it is up to the server implementation if the timeout is honored.
 /// - To bound server resources, server should implement own timeout for incoming requests.
 #[async_trait]
 pub(crate) trait NetworkClient: Send + Sync + 'static {
+    // Whether the network client streams blocks to subscribed peers.
+    const SUPPORT_STREAMING: bool;
```

Comment:
I may just be missing it but how is this used to gate streaming?

Reply:
Yeah this is unused for this PR, but will be used in the follow up to decide if new blocks are propagated with Broadcaster or another mechanism.